src/plugin/file/zip/zipparser.js

```diff
@@ -0,0 +1,379 @@
+/**
+ * @inheritDoc
+ */
+plugin.file.zip.ZIPParser.prototype.createZipReader = function(source) {
+  this.processingZip_ = true;
+  plugin.file.zip.ZIPParser.base(this, 'createZipReader', source);
+};
+
+
+/**
+ * @inheritDoc
+ */
+plugin.file.zip.ZIPParser.prototype.handleZipEntries = function(entries) {
+  if (typeof entries == 'undefined' || entries === null || entries.length == 0) {
+    this.logWarning_('No file(s) found in ZIP!');
+  } else {
+    this.semaphore_ = true;
+
+    for (var i = 0, n = entries.length; i < n; i++) {
+      this.zipEntries_++;
+
+      var entry = entries[i];
+
+      // build an "entry" object, detect filetype mime, etc
+      entry.getData(new zip.ArrayBufferWriter(), this.processZIPEntry_.bind(this, entry));
+    }
+
+    this.semaphore_ = false;
+  }
+};
+
+
+/**
+ * @param {zip.Entry} entry
+ * @param {*} content
+ * @private
+ */
+plugin.file.zip.ZIPParser.prototype.processZIPEntry_ = function(entry, content) {
+  if (content && content instanceof ArrayBuffer) {
+    var self = this;
+
+    var callback = function(uio) {
+      if (uio) {
+        var reader = new FileReader();
+        reader.onload = self.handleZIPText_.bind(self, uio);
+        reader.readAsText(new Blob([content]));
+      } else {
+        // couldn't determine file type OR is a directory
+        self.onComplete_(null);
+      }
+    };
+
+    this.toUIO_(entry, content, callback);
+  } else {
+    this.logError_('There was a problem unzipping the file!');
+    this.onError();
+  }
+};
+
+
+/**
+ * @param {osx.import.FileWrapper} uio
+ * @param {Event} event
+ * @private
+ */
+plugin.file.zip.ZIPParser.prototype.handleZIPText_ = function(uio, event) {
+  var content = (event && event.target) ? event.target.result : null;
+
+  if (content && typeof content === 'string') {
+    if (uio) uio['file'].setContent(content);
+    this.onComplete_(uio);
+  } else {
+    this.logError_('There was a problem reading the ZIP content!');
+    this.onError();
+  }
+};
+
+
+/**
+ * Create a UI Object to which Angular can bind form elements
+ *
+ * @param {Object} entry
+ * @param {ArrayBuffer} content
+ * @param {Function} callback
+ * @private
+ */
+plugin.file.zip.ZIPParser.prototype.toUIO_ = function(entry, content, callback) {
+  if (!entry || !entry.filename || !content || entry.directory) {
+    if (callback) callback(null);
+    return;
+  }
+
+  var file = new os.file.File();
+  file.setFileName(entry.filename);
+  file.setUrl('local://' + entry.filename);
+
+  var onDetect = function(type) {
+    if (type) {
+      var chain = os.file.mime.getTypeChain(type);
+      if (chain && chain.indexOf(os.file.mime.text.TYPE) > -1) {
+        file.convertContentToString();
+      }
+
+      file.setContentType(type);
+      file.setType(type);
+      return file;
+    }
+    return null;
+  };
+
+  var onFile = function(file) {
+    if (file) {
+      // turn this into a better object for the UI
+      return /** @type {osx.import.FileWrapper} */ ({
```

Comment:
Same here with the note about the quotes.

Reply:
Since this is an extern typedef, using quotes around the keys is unnecessary as the compiler already knows not to minify those symbols.

Reply:
Updated!  That'll save a some bytes.